pkg/okctlapplication/utils.go

```diff
@@ -0,0 +1,56 @@
+package okctlapplication
+
+import (
+	"os"
+
+	"github.com/oslokommune/okctl/pkg/config"
+	"github.com/oslokommune/okctl/pkg/config/load"
+	"github.com/oslokommune/okctl/pkg/config/state"
+	"github.com/oslokommune/okctl/pkg/okctl"
+	"github.com/spf13/cobra"
+)
+
+func loadRepoData(o *okctl.Okctl, cmd *cobra.Command) error {
+	repoDataNotFound := load.CreateOnRepoDataNotFound()
+
+	if o.NoInput {
+		repoDataNotFound = load.ErrOnRepoDataNotFound()
+	}
+
+	o.RepoDataLoader = load.RepoDataFromConfigFile(cmd, repoDataNotFound)
+
+	return o.LoadRepoData()
+}
+
+/*
+GetCluster returns a certain cluster if env is specified and DefaultRepositoryConfig is available.
+If no env is specified and DefaultRepositoryConfig is available, GetCluster will return the first
+cluster it finds.
+
+If theres no DefaultRepositoryConfig available, GetCluster returns nil
+*/
+func GetCluster(o *okctl.Okctl, cmd *cobra.Command, env string) *state.Cluster {
```

Comment:
We have the `RepoStateFromEnv` that contains this data: https://github.com/oslokommune/okctl/blob/master/pkg/config/config.go#L165, you can pass this in to your struct, using the specific interface you need, e.g., `Clusterer` (https://github.com/oslokommune/okctl/blob/master/pkg/config/state/repositorysaver.go#L20)